src/fields/abstractField.js

```diff
@@ -72,6 +84,34 @@ export default {
 	},
 
 	methods: {
+		deleteDataOnHide() {
+			let field = this.schema;
+			let visible = field.visible;
+
+			if (isFunction(field.visible)) {
+				visible = field.visible.call(this, this.model, field, this);
+			}
+
+			if (isNil(field.visible)) {
+				visible = true;
+			}
```

Comment:
I think we can make this portion of visibility checking a util function that both formgroup and abstractfield use. Just so if we ever need to change visibility logic we do it in one place.